src/node.cc

```diff
@@ -2497,36 +2501,95 @@ struct node_module* get_linked_module(const char* name) {
   return mp;
 }
 
-// DLOpen is process.dlopen(module, filename).
+struct DLib {
+  void* handle_;
+  std::string filename_;
+  std::string errmsg_;
+  int flags_;
+
+  void* Handle(struct DLib *lib) {
```

Comment:
That's an... unusual idiom.  Make it static (but why?) or drop the parameter and do `return handle_;` directly.

Reply:
Hm, to be honest, I wasn't familiar at all with C++ structs and I assumed they worked like C structs. Shame on me.